rct229/rules/section5/section5rule49.py

```diff
@@ -0,0 +1,146 @@
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+    get_surface_conditioning_category_dict,
+)
+from rct229.ruleset_functions.get_zone_conditioning_category_dict import (
+    ZoneConditioningCategory as ZCC,
+    get_zone_conditioning_category_dict,
+)
+
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ONE, ZERO
+from rct229.utils.std_comparisons import std_equal
+
+
+class Section5Rule49(RuleDefinitionListIndexedBase):
+    """Rule 49 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule49, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule49.BuildingRule(),
+            index_rmr="proposed",
+            id="5-49",
+            description="The proposed air leakage rate of the building envelope (I75Pa) at a fixed building pressure differential of 0.3 in. of water shall be 0.6 cfm/ft2 for buildings providing verification in accordance with Section 5.9.1.2. The air leakage rate of the building envelope shall be converted to appropriate units for the simulation program using one of the methods in Section G3.1.1.4. Exceptions: When whole-building air leakage testing, in accordance with Section 5.4.3.1.1, is specified during design and completed after construction, the proposed design air leakage rate of the building envelope shall be as measured..",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):  # put it under the "BuildingRule"
+        rmr_proposed = context.proposed
+        return {"climate_zone": rmr_proposed["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule49.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                each_rule=Section5Rule49.BuildingRule.ZoneRule(),
+                index_rmr="proposed",
+                list_path="$..zones[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_p = context.proposed
+
+            return {
+                **data,
+                "scc_dict_p": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building_p
+                ),
+                "zcc_dict_p": get_zone_conditioning_category_dict(
+                    data["climate_zone"], building_p
+                ),
+            }
+
+        class ZoneRule(RuleDefinitionBase):  # change to the zoneRule
```

Comment:
Because the `building_total_measured_air_leakage_rate` needs to accumulate at the building level, and because the `empty_measrued_air_leakage_rate_flow_flag` needs to be set at the building level, I think we do not want a `ZoneRule` for this one. Instead, we should loop of the zones inside `get_calc_vals` in BuildingRule.